src/store/store.ts

```diff
@@ -14,3 +15,5 @@ export type RootState = ReturnType<typeof store.getState>;
 export type AppDispatch = typeof store.dispatch;
 
 export type AppThunk<ReturnType = void> = ThunkAction<ReturnType, RootState, unknown, Action<string>>;
+
+export type ArrayElementType<T> = T extends (infer E)[] ? E : never;
```

Comment:
You don't need this if you restructure your reducer

Reply:
I couldn't find another way except this